src/main/java/seedu/address/commons/core/Messages.java

```diff
@@ -9,5 +14,21 @@ public class Messages {
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
+    public static final String MESSAGE_DUPLICATE_FIELDS = "Detected multiple values for the following field%1$s: "
+            + "%2$s.\nOnly one value is allowed for %3$s field%1$s\n";
+
+    /**
+     * Formats the duplicate prefixes into an error message.
+     */
+    public static String getDuplicatePrefixesToMessage(Set<Prefix> duplicatePrefixes) {
+        if (duplicatePrefixes.size() == 0) {
+            return "";
+        }
+
+        Set<String> duplicateFields = duplicatePrefixes.stream().map(Prefix::toString).collect(Collectors.toSet());
+
+        return String.format(MESSAGE_DUPLICATE_FIELDS, duplicateFields.size() > 1 ? "s" : "",
+                String.join(" ", duplicateFields), duplicateFields.size() > 1 ? "these" : "this");
+    }
```

Comment:
I think we are complicating this code too much for too little payback. No need to bother with singular/plural differences. Something like the following?

`Multiple values specified for the following single-valued filed(s): ...`

Reply:
Personally, I think over complicating this code is okay here since everything related to the messages have been abstracted to within `Messages`. So, there won't be any additional code duplication and its self-contained to within `Messages`.

That being said, I think changing it the message to
```
Multiple values specified for the following single-valued field(s): ...
Please only supply 1 value for the field(s) mentioned above.
```
can make it appear more user-friendly?

Reply:
>Personally, I think over complicating this code is okay here since everything related to the messages have been abstracted to within Messages. So, there won't be any additional code duplication and its self-contained to within Messages.

**Always** be as economical as possible with code. Every line of code is an additional burden every project member has to carry for the rest of the lifetime of the project.

>That being said, I think changing it the message to
>
>Multiple values specified for the following single-valued field(s): ...
Please only supply a single value for the field(s) mentioned above.
can make it appear more user-friendly?

Not totally opposed, but I have some concerns:
1. Feels a bit redundant. Check how the UI present this error to the user. Perhaps the first sentence is enough for the user to understand the problem.
2. Check other error messages. If they don't have `Please do such and such ...` part, we don't want to make this specific error message different from others.